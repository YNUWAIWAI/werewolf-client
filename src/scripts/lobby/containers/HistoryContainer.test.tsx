import * as ActionTypes from '../constants/ActionTypes'
import * as React from 'react'
import History from '../components/templates/History'
import HistoryContainer from './HistoryContainer'
import IntlProviderContainer from './IntlProviderContainer'
import {Provider} from 'react-redux'
import fakeStore from './fakeStore'
import {mount} from 'enzyme'

describe('<HistoryContainer />', () => {
  test('render', () => {
    const store = fakeStore()
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <HistoryContainer />
        </IntlProviderContainer>
      </Provider>
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
  test('selectVillage', () => {
    const store = fakeStore()
    const dispatch = jest.fn()

    store.dispatch = dispatch
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <HistoryContainer />
        </IntlProviderContainer>
      </Provider>
    )
    const id = 1

    wrapper.find(History).props().selectVillage(id)()
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch).toHaveBeenCalledWith({
      id,
      type: ActionTypes.App.SELECT_VILLAGE
    })
  })
})
