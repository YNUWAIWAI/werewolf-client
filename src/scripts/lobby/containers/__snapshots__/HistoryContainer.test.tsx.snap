// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoryContainer /> render 1`] = `"<div class=\\"lo--grid\\"><div class=\\"lo--header\\">History</div><div class=\\"lo--main-content\\"><div class=\\"lo--village\\"></div></div><div class=\\"lo--aside-content \\"><ul class=\\"lo--compact-menu\\"><li class=\\"lo--compact-menu--item\\"><button class=\\"lo--compact-menu--item--button \\" tabindex=\\"-1\\">Return to the main page</button></li></ul></div></div>"`;
